packages/imt.sol/contracts/internal/InternalLeanIMT.sol

```diff
@@ -73,6 +73,132 @@ library InternalLeanIMT {
         return node;
     }
 
+    /// @dev Inserts many leaves into the incremental merkle tree.
+    /// The function ensures that the leaves are valid according to the
+    /// constraints of the tree and then updates the tree's structure accordingly.
+    /// @param self: A storage reference to the 'LeanIMTData' struct.
+    /// @param leaves: The values of the new leaves to be inserted into the tree.
+    /// @return The root after the leaves have been inserted.
+    function _insertMany(LeanIMTData storage self, uint256[] calldata leaves) internal returns (uint256) {
+        // Check that all the new values are correct to be added.
+        for (uint256 i = 0; i < leaves.length; ) {
+            if (leaves[i] >= SNARK_SCALAR_FIELD) {
+                revert LeafGreaterThanSnarkScalarField();
+            } else if (leaves[i] == 0) {
+                revert LeafCannotBeZero();
+            } else if (_has(self, leaves[i])) {
+                revert LeafAlreadyExists();
+            }
+
+            self.leaves[leaves[i]] = self.size + i;
```

Comment:
Is `self.size + 1` always the same value here? Looks like its value is not incremented.

Reply:
Hey! It is `i` not `1`

Reply:
> Hey! It is `i` not `1`

Right!

Reply:
Yes, @0xjei it is correct.